Assets/Scripts/Unity/InitScript.cs

```diff
@@ -0,0 +1,110 @@
+/// </summary>
+public class InitScript : MonoBehaviour
+{
+    /// <summary>
+    ///  Called when the game starts.
+    /// </summary>
+    void Start()
+    {
+        this.AddControllers();
+
+        // Todo initialise localiser
+        AbstractUserLocalizer localizer = null;
+        this.InitControllers(localizer);
+    }
+
+    /// <summary>
+    ///  Add the controllers to the GameObject.
+    /// </summary>
+    private void AddControllers()
+    {
+        IEnumerable<AbstractSensorController> sensorControllers = this.GetAbstractControllers();
+        foreach (AbstractSensorController controller in sensorControllers)
+        {
+            gameObject.AddComponent(controller.GetType());
+        }
+    }
+
+    /// <summary>
+    ///  Get all the Scripts that inherit AbstractSensorController.
+    /// </summary>
+    /// <returns>List of the controller class types that inherit AbstractSensorController</returns>
+    private IEnumerable<AbstractSensorController> GetAbstractControllers()
+    {
+        return typeof(AbstractSensorController)
+            .Assembly.GetTypes()
+            .Where(t => t.IsSubclassOf(typeof(AbstractSensorController)) && !t.IsAbstract)
+            .Select(t => (AbstractSensorController)Activator.CreateInstance(t));
+    }
+
+    /// <summary>
+    ///  Initialize the controllers and register them to a localizer.
+    /// </summary>
+    /// <param name="localizer">The Localizer filter</param>
+    private void InitControllers(AbstractUserLocalizer localizer)
+    {
+        AbstractSensorController[] controllers = gameObject.GetComponents<AbstractSensorController>();
+        for (int i = 0; i < controllers.Length; i++)
+        {
+            controllers[i].Init();
+            this.RegisterLocationReceiver(localizer, controllers[i].GetLocationSource());
+            this.RegisterMotionReceiver(localizer, controllers[i].GetMotionSource());
+        }
+    }
+
+    /// <summary>
+    /// Registers a location source if the localizer can register it.
+    /// </summary>
+    /// <param name="localizer">The AbstractUserLocalizer</param>
+    /// <param name="source">The location source to register</param>
+    private void RegisterLocationReceiver(AbstractUserLocalizer localizer, ILocationSource source)
+    {
+        if (localizer is ILocationReceiver && source != null)
+        {
+            (localizer as ILocationReceiver).RegisterLocationReceiver(source);
+        }
+    }
+
+    /// <summary>
+    /// Registers a motion source if the localizer can register it.
+    /// </summary>
+    /// <param name="localizer">The AbstractUserLocalizer</param>
+    /// <param name="source">The motion source to register</param>
+    private void RegisterMotionReceiver(AbstractUserLocalizer localizer, IMotionSource source)
+    {
+        if (localizer is IMotionReceiver && source != null)
+        {
+            (localizer as IMotionReceiver).RegisterMotionSource(source);
```

Comment:
I did oversee that, but my SEM-senses are still tingling here since we are using the c sharp equivalent of java's instanceof. I'm not sure what the better solution here is though, if there even is one. What do you think @RemcodeVos ?